vitrina/datasets/views.py

```diff
@@ -175,12 +177,43 @@ def get_context_data(self, **kwargs):
             'tags': dataset.get_tag_list(),
             'subscription': [],
             'rating': 3.0,
-            'status': dataset.get_status_display()
+            'status': dataset.get_status_display(),
+            'resources': dataset.datasetdistribution_set.all(),
         }
         context_data.update(extra_context_data)
         return context_data
 
 
+class DatasetDistributionDownloadView(View):
+    def get(self, request, organization_kind, organization_slug, dataset_slug, pk, filename):
+        distribution = get_object_or_404(
+            DatasetDistribution,
+            dataset__organization__kind=organization_kind,
+            dataset__organization__slug=organization_slug,
+            dataset__slug=dataset_slug,
+            pk=pk,
+            filename__icontains=filename
+        )
+        response = FileResponse(open(distribution.filename.path, 'rb'))
+        return response
+
+
+class DatasetDistributionPreviewView(View):
+    def get(self, request, dataset_id, distribution_id):
+        distribution = get_object_or_404(
+            DatasetDistribution,
+            dataset__pk=dataset_id,
+            pk=distribution_id
+        )
+        data = []
+        if distribution.filename:
+            try:
```

Comment:
```suggestion
        if distribution.is_previewable():
```

Be consistent with what you do in templates. Don't just try to preview anything and swallow all the errors silently.